src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -212,84 +224,109 @@ const installDistributables = async (gamePath: string) => {
 
 async function downloadGame(
   appName: string,
-  downloadPath: string,
+  directory: string,
+  fileName: string,
   platformInfo: PlatformConfig,
   destinationPath: string
 ): Promise<void> {
-  let downloadStarted = false
+  /* eslint-disable-next-line no-async-promise-executor */
+  return new Promise(async (res, rej) => {
+    let downloadStarted = false
 
-  logInfo(`Downloading zip file to ${downloadPath}`, LogPrefix.HyperPlay)
+    logInfo(
+      `Downloading zip file to directory ${directory} filename ${fileName}`,
+      LogPrefix.HyperPlay
+    )
 
-  // we might need a helper function to deal with the different platforms
-  const window = getMainWindow()
+    // we might need a helper function to deal with the different platforms
+    const window = getMainWindow()
 
-  if (!window || !platformInfo.external_url) {
-    throw new Error('DownloadUrl not found')
-  }
+    if (!window || !platformInfo.external_url) {
+      throw new Error('DownloadUrl not found')
+    }
 
-  const downloadUrl =
-    process.env.CI &&
-    process.env.MOCK_DOWNLOAD_URL &&
-    process.env.CI === 'e2e' &&
-    process.env.APP_NAME_TO_MOCK &&
-    process.env.APP_NAME_TO_MOCK === appName
-      ? process.env.MOCK_DOWNLOAD_URL
-      : platformInfo.external_url
-  logInfo(`Downloading from ${downloadUrl}`, LogPrefix.HyperPlay)
-  await downloadFile(
-    downloadUrl,
-    downloadPath,
-    createAbortController(appName),
-    (downloadedBytes, downloadSpeed, diskWriteSpeed, progress) => {
-      const eta = calculateEta(
-        downloadedBytes,
-        downloadSpeed,
-        Number.parseInt(platformInfo.downloadSize ?? '0')
-      )
+    function cleanUpDownload() {
+      inProgressDownloadsMap.delete(appName)
+      deleteAbortController(appName)
+      rmSync(directory, { recursive: true, force: true })
+    }
 
-      if (downloadedBytes > 0 && !downloadStarted) {
-        downloadStarted = true
-        sendFrontendMessage('gameStatusUpdate', {
+    const downloadUrl =
```

Comment:
Would be better to put this into a helper method to make this more cleaner.

Reply:
done